Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+        // TODO: nullability
+        return document.getElementById(contextID)! && $(document.getElementById(contextID)!.firstElementChild!);
+      };
+    }
+  }
+
+  private static ApplyPosition (popUpDiv: JQuery, clickEvent: Nullable<JQueryMouseEventObject>, referenceElement: JQuery): void
+  {
+    var space_top = Math.round(referenceElement.offset().top - $(document).scrollTop());
+    var space_bottom = Math.round($(window).height() - referenceElement.offset().top - referenceElement.height() + $(document).scrollTop());
+    var space_left = referenceElement.offset().left;
+    var space_right = $(window).width() - referenceElement.offset().left - referenceElement.width();
+
+    // position drop-down list
+    var top = clickEvent ? clickEvent.clientY : Math.max(0, space_top + referenceElement.outerHeight());
+    var left = clickEvent ? clickEvent.clientX : 'auto';
+    var right = clickEvent ? 'auto' : Math.max(0, $(window).width() - referenceElement.offset().left - referenceElement.outerWidth());
+
+    popUpDiv.css('top', top);
+    popUpDiv.css('bottom', 'auto');
+    popUpDiv.css('right', right as string);
+    popUpDiv.css('left', left as string);
+    popUpDiv.css('position', 'fixed');
+
+    // move dropdown if there is not enough space to fit it on the page
+    if ((popUpDiv.width() > space_left) && (space_left < space_right))
+    {
+      if (popUpDiv.offset().left < 0)
+      {
+        left = Math.max(0, referenceElement.offset().left);
+        popUpDiv.css('left', left);
+        popUpDiv.css('right', 'auto');
+      }
+    }
+    if (popUpDiv.height() > space_bottom)
+    {
+      if (popUpDiv.height() > $(window).height())
+      {
+        popUpDiv.css('top', 0);
+      }
+      else if (space_top > popUpDiv.height())
+      {
+        var bottom = Math.max(0, $(window).height() - referenceElement.offset().top - (referenceElement.outerHeight() - referenceElement.height()));
+
+        popUpDiv.css('top', 'auto');
+        popUpDiv.css('bottom', bottom);
+      }
+      else
+      {
+        popUpDiv.css('top', 'auto');
+        popUpDiv.css('bottom', 0);
+      }
+    }
+  }
+
+  public static ClosePopUp (updateFocus: boolean): void
+  {
+    if (DropDownMenu._blurTimer !== null)
+      clearTimeout(DropDownMenu._blurTimer);
+
+    if (DropDownMenu._currentPopup !== null
+        && document.body.contains(DropDownMenu._currentPopup))
+    {
+      const menuPopup = DropDownMenu._currentPopup;
+      const $menuButton = $('a[aria-controls="' + menuPopup.id + '"]');
+      $menuButton.unbind('focus', DropDownMenu.BlurHandler);
+      $menuButton.unbind('blur', DropDownMenu.BlurHandler);
+      const menuButton = $menuButton[0];
+      menuButton.setAttribute('aria-expanded', 'false');
+      menuButton.removeAttribute('aria-controls');
+      if (updateFocus === DropDownMenu._updateFocus)
+        menuButton.focus();
+      // TODO: missing nullcheck?
```

Comment:
I don't think that this warrants an own issue. We can assume that an element has a parent and just use `!` BANG